### PR TITLE
feature: disable mouse event when building windows on Linux, closes #5913

### LINKED ISSUE #5913: [feat] Disable mouse event when building windows on Linux
### Describe the problem

On linux, there might have burst mouse events periodically and this could block gtk's main eventloop.
For more information, please see https://github.com/tauri-apps/tao/issues/635

### Describe the solution you'd like

With latest tao patch, we can call [with_cursor_moved_event](https://docs.rs/tao/latest/tao/platform/unix/trait.WindowBuilderExtUnix.html#tymethod.with_cursor_moved_event) and set it to false as workaround. Mouse event isn't useful since webview itself can already detect such event.

### Alternatives considered

Other alternatives are described in https://github.com/tauri-apps/tao/issues/635.
Please comment there if you have more ideas.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shogohida**
Signed-off-by: Shogo Hida <shogo.hida@gmail.com>

Fixes #5913 

### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: core/tauri-runtime-wry/src/lib.rs [hunk 1]
```diff
@@ -722,7 +722,8 @@ impl WindowBuilder for WindowBuilderWrapper {
       .always_on_top(config.always_on_top)
       .content_protected(config.content_protected)
       .skip_taskbar(config.skip_taskbar)
-      .theme(config.theme);
+      .theme(config.theme)
+      .disable_mouse_event(config.disable_mouse_event);
```
Comment (wusyong) [inline core/tauri-runtime-wry/src/lib.rs]: You should create a target specific block like below to call this method.

File: core/tauri-runtime-wry/src/lib.rs [hunk 2]
```diff
@@ -950,6 +951,12 @@ impl WindowBuilder for WindowBuilderWrapper {
   fn get_menu(&self) -> Option<&Menu> {
     self.menu.as_ref()
   }
+
+  #[cfg(target_os = "linux")]
+  fn disable_mouse_event(mut self, disable: bool) -> Self {
```
Comment (wusyong) [inline core/tauri-runtime-wry/src/lib.rs]: After thinking about it, I think there's no reason to expose a method for it. We should just disable it in `with_config`.

Comment (shogohida) [reply core/tauri-runtime-wry/src/lib.rs]: Thanks for your comment! I made some changes according to it. Is my understanding correct? If not, please give me some suggestion of the code

https://github.com/tauri-apps/tauri/pull/6025/commits/5d1e22a28cb011771f8cac9eac034549e3169400

Comment (wusyong) [reply core/tauri-runtime-wry/src/lib.rs]: Sorry this is not what I mean. I should comment in `config.rs` instead.
What you should do is:
- Remove `disable_mouse_event` in config.rs
- Bring this method back
- Add this trait method in `core/tauri-runtime/src/webview.rs`

And the code block you added should work.
Please at least run the build once on your device.
Also you need to add a change file to explain the change.

**Review by wusyong — DISMISSED** [review]

**Review by wusyong — DISMISSED** [review]

**Comment by wusyong** [comment]
@lucasfernog CI failed with Windows builds which seems to be irrelevant to this PR. Could I merge this one?

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**